### ISSUE TITLE
Updating smartobject content does not update bounds

### ISSUE THREAD

**Issue by iwehrman** [adobe-photoshop/generator-core#305]
As reported here: https://groups.google.com/forum/#!topic/adobe-generator-dev/X1m3r6WgpUI

---

If you update the pixel content of a smart object, it does not update the bounds of the layer even if you re-fetch whole document data. Smartobject doc (the .psb) gets correct bounds, but the original/container doc gets original bounds information. I.e. when you draw to the smart object and save changes  you get 

Smartobject change event
```javascript
{
  "version":"1.2.0",
  "timeStamp":1434358334.922,
  "count":4,
  "id":1554,
  "parentID":1537,
  "layers":[{
    "id":6,
    "bounds":{
      "top":79,
      "left":16,
      "bottom":639,
      "right":400
      },
    "pixels":true
    }]
  }
```

Container doc change event:
```javascript
{
  "version":"1.2.0",
  "timeStamp":1434358340.415,
  "count":10,
  "id":1537,
  "layers":[{
    "id":5,
    "changed":true
    }]
  }
```

Get document data for container 
```javascript
{
  "id":5,
  "index":1,
  "type":"layer",
  "name":"doc_layer",
  "bounds":{
    "top":0,
    "left":0,
    "bottom":0,
    "right":0
    },
  "visible":true,
  "clipped":false,
  "generatorSettings":false,
  "smartObject":{
    "ID":"d453525d-133b-11e5-9de8-e18913d3f768",
    "placed":"d4563de7-133b-11e5-9de8-e18913d3f768",
    "antiAliasType":16,
    "type":2,
    "transform":[0,0,480,0,480,800,0,800],
    "comp":-1
    }
  }
```

Steps to replicate:

1. Create new document
2. Right click on the layer and convert it to a smart object
3. Open smartobject by double clicking the layer icon
4. Draw on layer, save and close
5. Get document data

**Comment by joelrbrandt** [comment]
Same song, different verse.

User reports same behavior in CC 2015.

@jaredadobe any chance this is related to #307 and #306 ? If so, want to go for the hat trick?

cc @timothynoel on this bug, as well as #307 and #306. Seems like all of these are on the native side.

Let me know if handing this through Watson would be better. It would be great to get this fixed for the next release.

**Comment by joelrbrandt** [comment]
Filed Watson bug 4036322

**Comment by timothynoel** [comment]
Fixed for 16.1

**Comment by joelrbrandt** [comment]
User indeed reports this is fixed in 2015.1.1. Thanks @timothynoel
